rootfs/Dockerfile

```diff
@@ -9,21 +9,39 @@ RUN adduser --system \
 
 COPY requirements.txt /app/requirements.txt
 
-RUN buildDeps='gcc git libffi-dev libpq-dev python3-dev'; \
+RUN buildDeps='gcc git libffi-dev libpq-dev python3-dev python3-pip python3-wheel python3-setuptools'; \
     apt-get update && \
     apt-get install -y --no-install-recommends \
         $buildDeps \
+        sudo \
         libpq5 \
-        python3 \
-        sudo && \
+        python3-minimal \
+        # cryptography package needs pkg_resources
+        python3-pkg-resources && \
     ln -s /usr/bin/python3 /usr/bin/python && \
-    curl -sSL https://bootstrap.pypa.io/get-pip.py | python - pip==8.1.2 && \
     mkdir -p /configs && chown -R deis:deis /configs && \
-    pip install --disable-pip-version-check --no-cache-dir -r /app/requirements.txt && \
-    rm -rf /root/.cache/pip/* && \
+    pip3 install --disable-pip-version-check --no-cache-dir -r /app/requirements.txt && \
+    # cleanup
     apt-get purge -y --auto-remove $buildDeps && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/man /usr/share/doc
+    apt-get autoremove -y && \
+    apt-get clean -y && \
+    # package up license files if any
+    tar -czf /usr/share/copyrights.tar.gz /usr/share/common-licenses* /usr/share/doc/*/copyright && \
+    rm -rf \
+        /usr/share/doc \
+        /usr/share/man \
+        /usr/share/info \
+        /usr/share/locale \
+        /var/lib/apt/lists/* \
+        /var/log/* \
+        /var/cache/debconf/* \
+        /usr/share/common-licenses* \
+        /etc/systemd \
+        /lib/lsb \
+        /lib/udev \
+        /usr/lib/x86_64-linux-gnu/gconv/IBM* \
+        /usr/lib/x86_64-linux-gnu/gconv/EBC* && \
+    mkdir -p /usr/share/man/man{1..8}
```

Comment:
Dockerfile commands run under /bin/sh. `mkdir foo{1..8}` is bash syntax. Notice the difference between these two commands:

```
$ docker run ubuntu /bin/sh -c 'mkdir -p /usr/share/man/man{1..8} && ls /usr/share/man'
cs
da
de
es
fi
fr
hu
id
it
ja
ko
man1
man3
man5
man7
man8
man{1..8}
nl
pl
pt
pt_BR
ru
sl
sv
tr
zh_CN
zh_TW
$ docker run ubuntu /bin/bash -c 'mkdir -p /usr/share/man/man{1..8} && ls /usr/share/man'
cs
da
de
es
fi
fr
hu
id
it
ja
ko
man1
man2
man3
man4
man5
man6
man7
man8
nl
pl
pt
pt_BR
ru
sl
sv
tr
zh_CN
zh_TW
```